code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -5,6 +5,45 @@
 		Weaken(4)
 
 /mob/living/carbon/human/attack_hand(mob/living/carbon/human/M)
+	if(is_vampire(M) && M.a_intent == "harm" && M.zone_sel.selecting == "head" && !is_vampire(src)) //Allows vampires to drain blood from people

```

Comment:
Ok so, I've come up with a way you can do this without shitting up human ```attack_hand()```.
Give vampires a marshal-arts datum, focused around being a vamp, which includes all the draining junk.